### DISCUSSION TITLE
Options to control interface names given by Libvirt

### DISCUSSION THREAD

**Discussion by mazzy89** [vagrant-libvirt/vagrant-libvirt#1675] [Q&A]
I haven't found a way to control virtual interface name created by Libvirt. I have the following use case. A machine which has a physical interface `enp89s0` configured like so

```
[Match]
Name=enp89s0
[Network]
Bridge=br0
[BridgeVLAN]
VLAN=166
```

the interface is plugged into a bridge `br0` which has `VLANFiltering` set to `on`

```
[NetDev]
Name=br0
Kind=bridge

[Bridge]
VLANFiltering=yes
```

and the following configuration network configuration

```
[Match]
Name=br0

[Network]
DHCP=yes
```

At this point all the VMs created via vagrant-libvirt are placed into the VLAN=166 thanks to the following config

```
[Match]
Name=vnet*

[Network]
Bridge=br0

[BridgeVLAN]
VLAN=166
PVID=166
EgressUntagged=166
```

This design allow to have one unique bridge which filter VLANs.

however in this design I find a flaw. before launching the VM we do not know the name of the virtual interface so we need to match them all. In the case of having VMs running in a different VLAN then this would not be possible because even if we have a different bridge for control different VLAN traffic, still the match with the interface would go to select all the interfaces.
At this point then I'm wondering whether there is a way to control  the name of the virtual interface created by Libvirt.

**Comment by electrofelix** [comment]
I presume this is in relation to the devices creating on the host? I presume the issue is that it's a public network configuration that you are looking to define? otherwise should be able to set `:libvirt__iface_name => '<dev>'` for private networks.

I'm afraid I've had trouble getting the public network option to work out the box without a bridged device to use so I'm not really able to test whether exposing the same option in the https://github.com/vagrant-libvirt/vagrant-libvirt/blob/main/lib/vagrant-libvirt/templates/public_interface.xml.erb template would work as expected or cause other issues.

**Reply by mazzy89** [reply]
> I presume this is in relation to the devices creating on the host?

correct. it's a public network.

as I said in a previous comment, I'm going to set up a box using podman so I can in a much more flexible way help here to test stuff.

**Reply by electrofelix** [reply]
Looking around I've found #799 which is related to this and digging into the libvirt code I don't see any reason that setting `<target dev='...' />` would not be supported for any of the types, I think it might be worth me adding a PR to see if picking up `:libvirt__iface_name => '<dev>'` and setting the XML works for your case but only being set if the option is should limit risk of breakage.

**Comment by electrofelix** [comment]
@mazzy89 I think #1692 might solve this for you if you want to take it for a test to confirm. I do need to add some validation because the libvirt spec says that device names starting with specific prefixes will be ignored, e.g. vnet, but presumably you want to use a different naming scheme anyway.
